library/ssl_tls12_client.c

```diff
@@ -1774,19 +1774,9 @@ static int ssl_parse_server_ecdh_params(mbedtls_ssl_context *ssl,
         return MBEDTLS_ERR_SSL_DECODE_ERROR;
     }
 
-    /* When FFDH is enabled, the array handshake->xxdh_psa_peer_key size takes into account
-       the sizes of the FFDH keys which are at least 2048 bits.
-       The size of the array is thus greater than 256 bytes which is greater than any
-       possible value of ecpoint_len (type uint8_t) and the check below can be skipped.*/
-#if !defined(PSA_WANT_ALG_FFDH)
-    if (ecpoint_len > sizeof(handshake->xxdh_psa_peerkey)) {
-        return MBEDTLS_ERR_SSL_HANDSHAKE_FAILURE;
-    }
-#else
-    MBEDTLS_STATIC_ASSERT(sizeof(handshake->xxdh_psa_peerkey) >= UINT8_MAX,
-                          "peer key buffer too small");
-#endif
-
+    /* xxdh_psa_peerkey is sized after maximum supported FFDH public key.
+     * PSA_VENDOR_FFDH_MAX_KEY_BITS is always larger than EC public key, so
+     * we can skip the buffer size check before the memcpy-ing data into it. */
```

Comment:
This is correct now, but we intend to fix `PSA_EXPORT_PUBLIC_KEY_MAX_SIZE` so that it doesn't include the FFDH size when FFDH is not enabled, and then this will silently become incorrect.

Considering we're in a 1.2 file and this is only about ECC with PSA, I'd be inclined to add an unconditional check
```c
if (ecpoint_len >PSA_KEY_EXPORT_ECC_PUBLIC_KEY_MAX_SIZE(PSA_VENDOR_ECC_MAX_CURVE_BITS)) {
    return MBEDTLS_ERR_SSL_HANDSHAKE_FAILURE;
}
```
This is a bit stricter than necessary just to avoid an overflow in the next `memcpy()`, but OTOH: it's absolutely correct: if the length is larger than that, the peer's message was invalid so we should reject it (if we didn't reject it now, it would be rejected later when passed to `psa_raw_key_agreement()` anyway).

Also, it follows the simple rule than when we read a length from the outside world, the first thing we do is validate it. (Having a comment that explains why this is not necessary helps human readers, but not static analyzers.)